mon: add annotation to mon secret and cm (#9547)
rook/rook: Documentation/ceph-cluster-crd.md

You can set annotations / labels for Rook components for the list of key value pairs:

* `all`: Set annotations / labels for all components except `clusterMetadata`.
* `mgr`: Set annotations / labels for MGRs
* `mon`: Set annotations / labels for mons
* `osd`: Set annotations / labels for OSDs
* `prepareosd`: Set annotations / labels for OSD Prepare Jobs
* `monitoring`: Set annotations / labels for service monitor
* `crashcollector`: Set annotations / labels for crash collectors

> * `clusterMetadata`: Set annotations  only to `rook-ceph-mon-endpoints` configmap and the  `rook-ceph-mon` and `rook-ceph-admin-keyring` secrets. And `clusterMetadata` annotations will **not** be merged with `all` annotation. These annotations is applied only on critical resources. So, that these critical resources can be backed. Check [this issue](https://github.com/rook/rook/issues/8999) for more details.

Reviewer: In the docs we generally don't refer to github issues. If we need more details, it should just be in the docs.
```suggestion
* `clusterMetadata`: Set annotations  only to `rook-ceph-mon-endpoints` configmap and the  `rook-ceph-mon` and `rook-ceph-admin-keyring` secrets. These annotations will not be merged with the `all` annotations. The common usage is for backing up these critical resources with `kubed`.
```

> Example:

Reviewer: Let's remove this example from the doc, the cluster.yaml example should be sufficient.

You can set annotations / labels for Rook components for the list of key value pairs:

* `all`: Set annotations / labels for all components except `clusterMetadata`.
* `mgr`: Set annotations / labels for MGRs
* `mon`: Set annotations / labels for mons
* `osd`: Set annotations / labels for OSDs
* `prepareosd`: Set annotations / labels for OSD Prepare Jobs
* `monitoring`: Set annotations / labels for service monitor
* `crashcollector`: Set annotations / labels for crash collectors
* `clusterMetadata`: Set annotations  only to `rook-ceph-mon-endpoints` configmap and the  `rook-ceph-mon` and `rook-ceph-admin-keyring` secrets. These annotations will not be merged with the `all` annotations. The common usage is for backing up these critical resources with `kubed`.

> # And clusterMetadata annotation will **not** be merged with `all` annotation.

Reviewer: I don't think we want # for a header :), also remove the blank line so it's in the same paragraph with clusterMetadata.
```suggestion
Note the clusterMetadata annotation will not be merged with the `all` annotation.
```

Author: updated
